From 33c7334a19d16495d6c1edcb84e3ff8f52a69722 Mon Sep 17 00:00:00 2001
From: Hyun Kwon <hyun.kwon@xilinx.com>
Date: Mon, 5 May 2014 19:52:32 -0700
Subject: [PATCH 0367/1566] drm: xylon: connector: Call the encoder slave's
 mode_valid() function

Call the mode valid function of the necoder slave, and let the encoder
slave driver check validity of given mode.

Signed-off-by: Hyun Kwon <hyunk@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
---
 drivers/gpu/drm/xylon/xylon_connector.c |   14 +++++++++++---
 1 files changed, 11 insertions(+), 3 deletions(-)

diff --git a/drivers/gpu/drm/xylon/xylon_connector.c b/drivers/gpu/drm/xylon/xylon_connector.c
index 196b5f8..a0b5cbc 100644
--- a/drivers/gpu/drm/xylon/xylon_connector.c
+++ b/drivers/gpu/drm/xylon/xylon_connector.c
@@ -55,10 +55,18 @@ static int xylon_drm_connector_get_modes(struct drm_connector *base_connector)
 static int xylon_drm_connector_mode_valid(struct drm_connector *base_connector,
 					  struct drm_display_mode *mode)
 {
-	if (mode->clock > CLOCK_HIGH_LIMIT)
-		return MODE_CLOCK_HIGH;
+	struct xylon_drm_connector *connector =
+		to_xylon_connector(base_connector);
+	struct drm_encoder *encoder = connector->encoder;
+	struct drm_encoder_slave *encoder_slave = to_encoder_slave(encoder);
+	const struct drm_encoder_slave_funcs *encoder_sfuncs =
+		encoder_slave->slave_funcs;
+	int ret = MODE_OK;
+
+	if (encoder_sfuncs->mode_valid)
+		ret = encoder_sfuncs->mode_valid(encoder, mode);
 
-	return MODE_OK;
+	return ret;
 }
 
 static struct drm_encoder *
-- 
1.7.5.4

